### PR TITLE
Adding constyle

### PR DESCRIPTION
**Pull request by abrahammurciano**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by abrahammurciano** [comment]
@conda-forge-admin please ping conda-forge/help-python

**Comment by abrahammurciano** [comment]
@conda-forge-admin please ping cona-forge/help-python

Hi, how long should I expect to wait until it's added?

File: recipes/constyle/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+test:
+  imports:
+    - constyle
+  requires:
+    - pip
+  commands:
+    - pip check
+
+about:
+  home: https://github.com/abrahammurciano/constyle
```
Comment (hmaarrfk) [inline recipes/constyle/meta.yaml:35]: This link is wrong. Should be https://github.com/abrahammurciano/python-constyle

File: recipes/constyle/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: ceb31c979b2f0ca279aa8b915085405926cf4f0325def271a0e0eb7a2c581096
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+  skip: True
```
Comment (hmaarrfk) [inline recipes/constyle/meta.yaml:16]:
```suggestion
```
You aren't building anything with this line in there.

File: recipes/constyle/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+about:
+  home: https://github.com/abrahammurciano/constyle
+  summary: 'A Python library to add style to your console'
+  description: |
+    A Python library to add style to your console.
+  license: GPL-3.0-only
+  license_family: GPL
+  license_file: LICENSE
+  doc_url: https://abrahammurciano.github.io/python-constyle/constyle
+  dev_url: https://github.com/abrahammurciano/constyle
```
Comment (hmaarrfk) [inline recipes/constyle/meta.yaml:43]: this link also seems wrong.

**Comment by hmaarrfk** [comment]
Can you confirm that you package has in fact, no dependencies?

File: recipes/constyle/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.7
+    - pip
+    - poetry
+  run:
+    - python >=3.7
```
Comment (hmaarrfk) [inline recipes/constyle/meta.yaml:23]:
```suggestion
    - python >=3.7
    - importlib-metadata
```

**Comment by abrahammurciano** [comment]
> Can you confirm that you package has in fact, no dependencies?

You're right, I was missing importlib metadata, sorry about that, thanks

**Comment by hmaarrfk** [comment]
I guess we need this. https://github.com/conda-forge/importlib_metadata-feedstock/pull/107

Yo ushould also add the same constraint to your package.

**Comment by hmaarrfk** [comment]
The one specified by pip check.

**Comment by abrahammurciano** [comment]
I'm not sure I understand... Which constraint do I need to add, and how?

File: recipes/constyle/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - importlib-metadata
```
Comment (hmaarrfk) [inline recipes/constyle/meta.yaml:24]:
```suggestion
    - importlib-metadata >=4.12.0,<5.0.0a
```

The logs failed with
```
+ pip check
constyle 0.2.0 has requirement importlib-metadata<5.0.0,>=4.12.0, but you have importlib-metadata 4.11.4.
Tests failed for constyle-0.2.0-pyhd8ed1ab_0.tar.bz2 - moving package to /home/conda/staged-recipes/build_artifacts/broken
```

I think the trailing `a` helps against pre-release builds.

**Comment by abrahammurciano** [comment]
Oh, I see, yes that makes sense, thanks

**Comment by hmaarrfk** [comment]
The thing is, your pip dependencies specify version 4.12. So you would need to rerelease a new version if in fact you only require 4.11. Otherwise pip check will fail.

While you don't need pip check at runtime, many do find it frustrating if it fails.

**Comment by abrahammurciano** [comment]
> The thing is, your pip dependencies specify version 4.12. So you would need to rerelease a new version if in fact you only require 4.11. Otherwise pip check will fail.
> 
> While you don't need pip check at runtime, many do find it frustrating if it fails.

Yeah I was just in the process of releasing a new version with lower requirements

**Comment by abrahammurciano** [comment]
@hmaarrfk I don't understand why the URL is not found. I can't access it manually either, despite the version being visible on PYPI. However the old URL does work.
https://pypi.io/packages/source/c/constyle/constyle-0.2.1.tar.gz

**Comment by hmaarrfk** [comment]
i think some CDNs have been down recently

**Comment by abrahammurciano** [comment]
@hmaarrfk I finally got it to work

**Comment by hmaarrfk** [comment]
Great thank you! you should get an invite to join the github conda-forge organization

**Pull request merged**